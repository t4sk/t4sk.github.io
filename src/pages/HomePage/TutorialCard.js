import React from "react"
import PropTypes from "prop-types"
import { Card, Image } from "semantic-ui-react"
import styles from "./TutorialCard.module.css"

export function TutorialCard(props) {
  const { title, logo, href } = props

  return (
    <a href={href}>
      <Card>
        <Image src={logo} wrapped ui={false} />
        <Card.Content>
          <Card.Header>
            <div className={styles.title}>{title}</div>
          </Card.Header>
        </Card.Content>
      </Card>
    </a>
  )
}

TutorialCard.propTypes = {
  title: PropTypes.string.isRequired,
  logo: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
}

export default TutorialCard
