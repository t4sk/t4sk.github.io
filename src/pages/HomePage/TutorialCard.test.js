import React from "react"
import { shallow } from "enzyme"
import { TutorialCard } from "./TutorialCard"

const props = {
  title: "Learn Solidity",
  logo: "path/to/logo",
  path: "/tutorials/learn-solidity",
}

beforeEach(() => {})

test("it renders", () => {
  const component = shallow(<TutorialCard {...props} />)

  expect(component).toMatchSnapshot()
})
