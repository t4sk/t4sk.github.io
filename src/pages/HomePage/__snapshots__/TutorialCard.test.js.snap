// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<a
  href="/tutorials/learn-solidity"
>
  <Card>
    <Image
      as="img"
      src="path/to/logo"
      ui={false}
      wrapped={true}
    />
    <CardContent>
      <CardHeader>
        <div
          className="title"
        >
          Learn Solidity
        </div>
      </CardHeader>
    </CardContent>
  </Card>
</a>
`;
